File: despatch.Domain/Models/Default/Request/ImageRequest.cs
using despatch.Domain.Enums;

namespace despatch.Domain.Models.Default.Request
{
    public class ImageRequest : BaseEntityRequest
    {
        public string Name { get; set; }
        public string Description { get; set; }
        public ImageType Type { get; set; }
        public bool IsMain { get; set; }

        public string DeviceUrlId { get; set; }
        public string SourseUrlId { get; set; }

        public virtual UrlRequest DeviceUrl { get; set; }
        public virtual UrlRequest SourceUrl { get; set; }
    }
}

File: despatch.WEB/app/src/swagger/models/image-request.ts
/* tslint:disable */
import { BaseEntityRequest } from './base-entity-request';
import { UrlRequest } from './url-request';
import { ImageType } from './image-type';
export interface ImageRequest extends BaseEntityRequest{
  description?: string;
  deviceUrl?: UrlRequest;
  deviceUrlId?: string;
  isMain: boolean;
  name?: string;
  sourceUrl?: UrlRequest;
  sourseUrlId?: string;
  type: ImageType;
}
